### PR TITLE
PIN-4020 Implemented Update of the Risk Analisys

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-4021](https://github.com/pagopa/interop-be-catalog-management/pull/228) and [PIN-3981](https://github.com/pagopa/interop-commons/pull/283) to compile

[PIN-3981]: https://pagopa.atlassian.net/browse/PIN-3981?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PIN-4021]: https://pagopa.atlassian.net/browse/PIN-4021?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by beetlecrunch — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -911,6 +911,60 @@ paths:
             application/problem+json:
               schema:
                 $ref: '#/components/schemas/Problem'
+  /eservices/{eServiceId}/riskanalysis/{riskAnalysisId}:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
  /eservices/{eServiceId}/riskAnalysis/{riskAnalysisId}:
```

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 2]
```diff
@@ -677,6 +677,32 @@ final case class ProcessApiServiceImpl(
...
+      organizationId   <- getOrganizationIdFutureUUID(contexts)
+      eServiceUuid     <- eServiceId.toFutureUUID
+      riskAnalysisUuid <- riskAnalysisId.toFutureUUID
+      catalogItem      <- catalogManagementService.getEServiceById(eServiceUuid)
+      _                <- isDraftEService(catalogItem)
+      _                <- isReceiveEService(catalogItem)
+      _                <- assertRequesterAllowed(catalogItem.producerId)(organizationId)
+      tenant           <- tenantManagementService.getTenantById(organizationId)
+      tenantKind       <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
+      _                <- isRiskAnalysisFormValid(seed.riskAnalysisForm.toTemplate, true)(tenantKind.toTemplate)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]:
```suggestion
      _                <- isRiskAnalysisFormValid(seed.riskAnalysisForm.toTemplate, schemaOnlyValidation = true)(tenantKind.toTemplate)
```

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ResponseHandlers.scala [hunk 3]
```diff
@@ -63,6 +63,22 @@ object ResponseHandlers extends AkkaResponses {
...
+  def updateRiskAnalysisResponse[T](logMessage: String)(
+    success: T => Route
+  )(result: Try[T])(implicit contexts: Seq[(String, String)], logger: LoggerTakingImplicit[ContextFieldsToLog]): Route =
+    result match {
+      case Success(s)                                => success(s)
+      case Failure(ex: OperationForbidden.type)      => forbidden(ex, logMessage)
+      case Failure(ex: EServiceNotFound)             => notFound(ex, logMessage)
+      case Failure(ex: EServiceRiskAnalysisNotFound) => notFound(ex, logMessage)
+      case Failure(ex: TenantNotFound)               => notFound(ex, logMessage)
+      case Failure(ex: TenantKindNotFound)           => notFound(ex, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ResponseHandlers.scala]:
```suggestion
```

File: src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala [hunk 4]
```diff
@@ -695,8 +695,9 @@ final case class ProcessApiServiceImpl(
       _                <- assertRequesterAllowed(catalogItem.producerId)(organizationId)
       tenant           <- tenantManagementService.getTenantById(organizationId)
       tenantKind       <- tenant.kind.toFuture(TenantKindNotFound(tenant.id))
-      _                <- isRiskAnalysisFormValid(seed.riskAnalysisForm.toTemplate, true)(tenantKind.toTemplate)
-      _                <- catalogManagementService.updateRiskAnalysis(eServiceUuid, riskAnalysisUuid, seed.toDependency)
+      _ <- isRiskAnalysisFormValid(seed.riskAnalysisForm.toTemplate, schemaOnlyValidation = true)(tenantKind.toTemplate)
+
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala]: 🧹

File: src/main/resources/interface-specification.yml (hunk 1 above)
Comment (galales) [reply src/main/resources/interface-specification.yml]: @nttdata-rtorsoli it seems this has not been updated

**Review by galales — APPROVED** [review]

**Pull request merged**